app/views/components/InputRadio.scala.html

```diff
@@ -14,30 +14,46 @@
  * limitations under the License.
  *@
 
-@import uk.gov.hmrc.govukfrontend.views.html.components.implicits._
+@import viewModels.components.InputRadioViewModel
+@import viewModels.components.InputRadioViewModel._
+@import views.html.components.Heading
 @import views.utils.ViewUtils._
 
-@this(govukRadios: GovukRadios)
+@this(govukRadios: GovukRadios, hmrcHeading: Heading)
 
 @(
     field: Field,
-    legend: String,
     caption: Option[String] = None,
+    radioType: InputRadioViewModel,
     hint: Option[String] = None,
     radioItems: String => Seq[RadioItem],
     inline: Boolean = false
 )(implicit messages: Messages)
 
-@govukRadios(
+@radios = @{
     Radios(
         idPrefix = Some(field.id),
         name = field.name,
-        classes = if (inline) "govuk-radios--inline" else "",
+        classes = if(inline) "govuk-radios--inline" else "",
         errorMessage = field.error.map { e =>
             ErrorMessage.errorMessageWithDefaultStringsTranslated(content = Text(messages(e.message, e.args: _*)))
         },
         hint = hint.map(x => Hint(content = Text(x))),
         items = radioItems(field.id)
     )
-    .withHeadingAndCaption(heading = legend, caption = caption)
-)
+}
+
+@radioType match {
+    case Radio(heading, caption) => {
+        @govukRadios(radios.withHeadingAndCaption(heading, caption))
+    }
+    case x @ RadioWithAdditionalHtml(heading, caption, additionalHtml) => {
+        @hmrcHeading(heading, caption, Some(x.headingClasses))
+        @additionalHtml
+        @govukRadios(radios.withLegend(heading, false))
+    }
+    case RadioWithLegend(legend) => {
```

Comment:
I wouldn’t anticipate this one ever being needed so can likely be removed (including the case class)